assignment1/isPermutation.py

```diff
@@ -0,0 +1,36 @@
+def isPermutation(str1, str2):
+    """Function that checks if one string is a permutation of the other.
+
+    Args:
+        str1: fist string;
+        str2: second string.
+
+    Return:
+        a bool, True if the first string is a permutation of the other,
+        and False otherwise.
+    """
+    if str1 is None or str2 is None:
+        return False
+    str1 = str1.lower()
+    str2 = str2.lower()
+    if str1 == str2:
+        return True
+    sybsFromStr1 = dict()
+    for symbol in str1:
+        sybsFromStr1[symbol] = sybsFromStr1.get(symbol, 0) + 1
+    for symbol in str2:
+        if symbol not in sybsFromStr1.keys():
```

Comment:
If `d` is a dictionary:
```python
s in d.keys()
```
and
```python
s in d
```
have the same meaning, i.e., checking if the symbol is in the keys is the same as checking if the symbol is in the dictionary.

You can simplify the `if`-statement here.